templates/repo/code/recently_pushed_new_branches.tmpl

```diff
@@ -5,7 +5,7 @@
 			{{$branchLink := HTMLFormat `<a href="%s/src/branch/%s">%s</a>` $.RepoLink (PathEscapeSegments .Name) .Name}}
 			{{ctx.Locale.Tr "repo.pulls.recently_pushed_new_branches" $branchLink $timeSince}}
 		</div>
-		<a role="button" class="ui compact positive button tw-m-0" href="{{$.Repository.ComposeBranchCompareURL $.Repository.BaseRepo .Name}}">
+		<a role="button" class="ui compact green button tw-m-0" href="{{$.Repository.ComposeBranchCompareURL $.Repository.BaseRepo .Name}}">
```

Comment:
Is it really right to do so? Especially for color-blindness theme?

If we use "positive" / "negative", it is easier to define colors for color-blindness , because these colors won't be green/red.

Reply:
Yes we can later refactor to success/warning/danger but I want to do this later.

https://shoelace.style/components/button#variants

Reply:
If it keeps "positive", the diff is truly minimal.

Reply:
I can rename "green" to "success", there's only 1 instance.

Reply:
Well I guess I may as well rename red now.

Reply:
What do you think about success/danger instead of positive/negative?

Reply:
such a rename would blow up the diff significantly because then I would also need to rename the color vars most likely. I'd rather defer this.

```css
  --color-error-border: #a04141;
  --color-error-bg: #522;
  --color-error-bg-active: #744;
  --color-error-bg-hover: #633;
  --color-error-text: #f9cbcb;
  --color-success-border: #458a57;
  --color-success-bg: #284034;
  --color-success-text: #6cc664;
  --color-warning-border: #bb9d00;
  --color-warning-bg: #3a3a30;
  --color-warning-text: #fbbd08;
  --color-info-border: #306090;
  --color-info-bg: #26354c;
  --color-info-text: #38a8e8;
```

Reply:
I prefer to keep a refactoring PR focused and avoid introduce unnecessary changes.

Reply:
So I can do the class rename but the css vars will stay at red/green. If those approve the class names above i will do it.

Reply:
My preference is to just keep the "positive / negative" names. Just make them have the styles as "green / red", and leave the renaming to the future.

Reply:
I prefer to not re-introduce positive/negative for now as I find success/danger to be better names. There are like 2 cases in the whole codebase that used positive/negative. green is one case, only red is much in use. So I prefer to keep red/green for now to keep the diff small and later refactor to success/danger, including color names.

Reply:
OK, I won't block this change.

But, "success" is a bad name to me for such button. "success" is a result, but doesn't really means what the button is for.

"danger" / "negative" either looks good to me.

Reply:
Yes, success is not a ideal word but "negative" also isn't. At least these frameworks use the suggested naming scheme:

https://shoelace.style/components/button#variants
https://bulma.io/documentation/elements/button/#colors